llvm-spirv/lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -223,8 +223,17 @@ static bool recursiveType(const StructType *ST, const Type *Ty) {
              StructTy->element_end();
     }
 
-    if (auto *PtrTy = dyn_cast<PointerType>(Ty))
-      return Run(PtrTy->getPointerElementType());
+    if (auto *PtrTy = dyn_cast<PointerType>(Ty)) {
+      auto *ElTy = PtrTy->getPointerElementType();
```

Comment:
AFAIK, according to LLVM guidelines, you shouldn't user auto here.

Reply:
Thanks, fixed